libraries/tools/kotlin-gradle-plugin/src/common/kotlin/org/jetbrains/kotlin/gradle/plugin/mpp/CompilationDetails.kt

```diff
@@ -250,7 +251,7 @@ open class DefaultCompilationDetails<T : KotlinCommonOptions>(
      * Adds `allDependencies` of configurations mentioned in `configurationNames` to configuration named [this] in
      * a lazy manner
      */
-    private fun String.addAllDependenciesFromOtherConfigurations(project: Project, vararg configurationNames: String) {
+    protected fun String.addAllDependenciesFromOtherConfigurations(project: Project, vararg configurationNames: String) {
         project.configurations.named(this).configure { receiverConfiguration ->
             receiverConfiguration.dependencies.addAllLater(
```

Comment:
side note: Can we use extendsFrom here?

Reply:
This comment here makes a point on why this might be undesirable: 
https://github.com/JetBrains/kotlin/blob/0c001cc9939a2ab11815263ed825c1096b3ce087/libraries/tools/kotlin-gradle-plugin/src/common/kotlin/org/jetbrains/kotlin/gradle/plugin/mpp/CompilationDetails.kt#L220

It claims that this is important for IDE resolution order. 
I am not sure if this still holds. Changing this would require re-investigating and testing